assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,307 @@
+            });
+          }          
+
+          if (buildFrom) {
+            $scope.buildFrom.imageStreams.push(key);
+            $scope.buildFrom.tags[key] = tagList;
+          } else {
+            $scope.pushTo.imageStreams.push(key);
+            $scope.pushTo.tags[key] = tagList;
+          }
+        });
+      });
+
+    }
+
+    $scope.updateTags = function(imageStream, buildFrom) {
+      if (buildFrom) {
+        $scope.options.pickedBuildFromImageStreamTag = "";
+      } else {
+        $scope.options.pickedPushToImageStreamTag = "";
+      }
+    }
+
+    $scope.save = function() {
+      // Update Configuration
+      $filter('buildStrategy')($scope.updatedBuildConfig).forcePull = $scope.options.forcePull;
+      if ($scope.updatedBuildConfig.spec.strategy.type === "Docker") {
+        $filter('buildStrategy')($scope.updatedBuildConfig).noCache = $scope.options.noCache;
+      }
+
+      var from = {}
+      if ($scope.options.pickedBuildFromType === "ImageStreamTag") {
+        from = {
+          kind: $scope.options.pickedBuildFromType,
+          namespace: $scope.options.pickedBuildFromNamespace,
+          name: $scope.options.pickedBuildFromImageStream + ":" + $scope.options.pickedBuildFromImageStreamTag
+        }
+      } else if ($scope.options.pickedBuildFromType === "DockerImage") {
+        from = {
+          kind: $scope.options.pickedBuildFromType,
+          name: $scope.options.pickedBuildFromDockerImage
+        }
+      }
+      $filter('buildStrategy')($scope.updatedBuildConfig).from = from;
+
+      var to = {
+        kind: "ImageStreamTag",
+        namespace: $scope.options.pickedPushToNamespace,
+        name: $scope.options.pickedPushToImageStream + ":" + $scope.options.pickedPushToImageStreamTag
+      }
+      $scope.updatedBuildConfig.spec.output.to = to;
+
+      // Update envVars
+      var updateEnvVars = [];
+      angular.forEach($scope.envVars, function(v, k) {
+        var env = {
+          name: k,
+          value: v
+        };
+        updateEnvVars.push(env);
+      });
+      $filter('buildStrategy')($scope.updatedBuildConfig).env = updateEnvVars
+
+      // Update triggers
+      var triggers = [];
+      if ($scope.triggers["webhook"]) {
+        angular.forEach($scope.buildConfig.spec.triggers, function(trigger) {
+          if (trigger.type === "GitHub") {
+            triggers.push(trigger);
+          }
+        });
+        if (triggers.length === 0) {
+          triggers.push(
+            {
+              github: {
+                secret: ApplicationGenerator._generateSecret()
+              },
+              type: "GitHub"
+            }
+          );
+        }
+      }
+      angular.forEach($scope.buildConfig.spec.triggers, function(trigger) {
+        if (trigger.type === "Generic") {
+          triggers.push(trigger);
+        }
+      });
+      if ($scope.triggers["imageChange"]) {
+        triggers.push({
+          imageChange: {},
```

Comment:
Are we potentially losing data here? It looks like there is a `lastTriggeredImageID` property.